src/Modules/Workspaces/Commands.Workspaces.Test/AddPowerBIWorkspaceUserTest.cs

```diff
@@ -0,0 +1,100 @@
+﻿/*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.PowerBI.Commands.Profile.Test;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.PowerBI.Commands.Workspaces.Test
+{
+    [TestClass]
+    public class AddPowerBIWorkspaceUserTest
+    {
+        [TestMethod]
+        [TestCategory("Interactive")]
+        [TestCategory("SkipWhenLiveUnitTesting")] // Ignore for Live Unit Testing
+        public void EndToEndAddPowerBIWorkspaceUserAsAdmin()
+        {
+            using (var ps = System.Management.Automation.PowerShell.Create())
+            {
+                ProfileTestUtilities.ConnectToPowerBI(ps);
+
+                var workspace = WorkspacesTestUtilities.GetWorkspace(ps);
+
+                if (workspace == null)
+                {
+                    Assert.Inconclusive("No workspaces found to perform end to end test");
+                }
+
+                var parameters = new Dictionary<string, object>()
+                {
+                    { "Scope", "Organization"},
+                    { "Id", workspace.Id},
+                    { "UserEmailAddress", "user1@granularcontrols1.ccsctp.net"}, //update parameters for all tests to use a test account, this user email will only work on Onebox
+                    { "UserAccessRight", "Admin" }
+                };
+
+                ps.AddCommand(new CmdletInfo($"{AddPowerBIWorkspaceUser.CmdletVerb}-{AddPowerBIWorkspaceUser.CmdletName}", typeof(AddPowerBIWorkspaceUser))).AddParameters(parameters);
+                var result = ps.Invoke();
+                Assert.IsNotNull(result);
+                Assert.IsTrue(result.Count > 0);
```

Comment:
It might be better to verify that the user was really added or removed from the workspace as I do in the Set-PowerBIWorkspace tests. I'm not sure checking the count on the result is too meaningful for an end-to-end test.

Reply:
We don't have the query parameters yet, so GetWorkspaces doesn't return the users. Also we don't have a cmdlet to get group users, GetWorkspaces as Individual doesn't return users. I will update the tests once we have this.